src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2321,6 +2321,46 @@ public static void TestNameWithInvariantCulture()
 
         }
 
+        [Theory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [InlineData("America/Buenos_Aires", "America/Argentina/Buenos_Aires")]
+        [InlineData("America/Catamarca", "America/Argentina/Catamarca")]
+        [InlineData("America/Cordoba", "America/Argentina/Cordoba")]
+        [InlineData("America/Jujuy", "America/Argentina/Jujuy")]
+        [InlineData("America/Mendoza", "America/Argentina/Mendoza")]
+        [InlineData("America/Indianapolis", "America/Indiana/Indianapolis")]
+        public static void TestTimeZoneIdBackwardCompatibility(string oldId, string currentId)
+        {
+            TimeZoneInfo oldtz = TimeZoneInfo.FindSystemTimeZoneById(oldId);
+            TimeZoneInfo currenttz = TimeZoneInfo.FindSystemTimeZoneById(currentId);
+
+            Assert.Equal(oldtz.StandardName, currenttz.StandardName);
+            Assert.Equal(oldtz.DisplayName, currenttz.DisplayName);
+        }
+
+        [Theory]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [InlineData("America/Buenos_Aires")]
+        [InlineData("America/Catamarca")]
+        [InlineData("America/Cordoba")]
+        [InlineData("America/Jujuy")]
+        [InlineData("America/Mendoza")]
+        [InlineData("America/Indianapolis")]
+        public static void ChangeLocalTimeZone(string id) 
```

Comment:
This test is checking that setting the `TZ` environment var leads to the correct tz being returned, is that correct? Unrelated to this PR?

Maybe rename the test to `ReadLocalTZFromEnvironmentVariable`

Reply:
Wasm currently uses uses the TZ setting via shared code with unix and this tests the exact case that was failing for browser so it is very related.

Reply:
Changing the `TZ` environment variable changes the local timezone. We're checking that when we change the local timezone with different ids that the correct one is still returned.